src/main/java/tc/oc/pgm/map/MapInfoImpl.java

```diff
@@ -0,0 +1,238 @@
+package tc.oc.pgm.map;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import java.util.Collection;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.TreeSet;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import javax.annotation.Nullable;
+import net.md_5.bungee.api.ChatColor;
+import org.apache.commons.lang3.builder.ToStringBuilder;
+import org.bukkit.Difficulty;
+import org.jdom2.Element;
+import tc.oc.component.Component;
+import tc.oc.component.types.PersonalizedText;
+import tc.oc.named.NameStyle;
+import tc.oc.pgm.AllTranslations;
+import tc.oc.pgm.api.map.Contributor;
+import tc.oc.pgm.api.map.MapInfo;
+import tc.oc.pgm.api.map.MapTag;
+import tc.oc.pgm.map.contrib.PlayerContributor;
+import tc.oc.pgm.map.contrib.PseudonymContributor;
+import tc.oc.pgm.util.TranslationUtils;
+import tc.oc.pgm.util.XMLUtils;
+import tc.oc.server.NullCommandSender;
+import tc.oc.util.Version;
+import tc.oc.xml.InvalidXMLException;
+import tc.oc.xml.Node;
+
+public class MapInfoImpl implements MapInfo {
+  private static final int DEFAULT_DIFFICULTY = Difficulty.NORMAL.ordinal();
+
+  private final String id;
+  private final Version proto;
+  private final Version version;
+  private final String name;
+  private final String description;
+  private final Collection<Contributor> authors;
+  private final Collection<Contributor> contributors;
+  private final Collection<String> rules;
+  private final int difficulty;
+  protected final Collection<MapTag> tags;
+  protected final Collection<Integer> players;
+
+  public MapInfoImpl(
+      @Nullable String id,
+      Version proto,
+      Version version,
+      String name,
+      String description,
+      @Nullable Collection<Contributor> authors,
+      @Nullable Collection<Contributor> contributors,
+      @Nullable Collection<String> rules,
+      @Nullable Integer difficulty,
```

Comment:
Why is this an integer instead of a Difficulty?

Reply:
See other comment.